sktime/transformations/series/window_summarizer.py

```diff
@@ -0,0 +1,357 @@
+        super(LaggedWindowSummarizer, self).__init__()
+
+    # Get extraction parameters
+    def _fit(self, X, y=None):
+        """Fit transformer to X and y.
+
+        Private _fit containing the core logic, called from fit
+        Parameters
+        ----------
+        X : pd.Series or pd.DataFrame
+        y : None
+
+        Returns
+        -------
+        X: pd.DataFrame
+        self: reference to self
+        """
+        # check if dict is empty
+
+        X_name = get_name_list(X)
+
+        if X_name is None:
+            self._X_rename = "var_0"
+            X_name = ["var_0"]
+        else:
+            self._X_rename = None
+
+        if y is not None:
+            y_name = get_name_list(y)
+            if y_name is None:
+                self._y_rename = "y"
+                y_name = ["y"]
+            else:
+                self._y_rename = None
+            Z_name = X_name + y_name
+        else:
+            Z_name = X_name
+            self._y_rename = None
+
+        if self.target_cols is None:
+            self._target_cols = [Z_name][0]
+        else:
+            self._target_cols = self.target_cols
+
+        if y is not None:
+            if not len(y_name + X_name) == len(set(y_name + X_name)):
+                raise ValueError(
+                    "Please make sure that names across X and y are not"
+                    + " duplicate. If unnamed X/y Series are provided, X will be"
+                    + " renamed to 'X_var_0' and y will be renamed to 'y'."
+                    + " This could also be result of this error if e.g. X"
+                    + " contained a named column y and an unnamed y Series"
+                    + " was renamed to 'y'."
+                )
+
+        if self.target_cols is not None:
+            if not all(x in Z_name for x in self.target_cols):
+                raise ValueError("target_X targets non-existing column in X")
+
+        if self.functions is None:
+            func_dict = pd.DataFrame(
+                {
+                    "lag": ["lag", [[1, 0]]],
+                }
+            ).T.reset_index()
+        else:
+            func_dict = pd.DataFrame(self.functions).T.reset_index()
+
+        func_dict.rename(
+            columns={"index": "name", 0: "win_summarizer", 1: "window"},
+            inplace=True,
+        )
+        func_dict = func_dict.explode("window")
+        self._func_dict = func_dict
+        self._truncate_start = func_dict["window"].apply(lambda x: x[0] + x[1]).max()
+
+        return self
+
+    def _transform(self, X, y=None):
+        """Transform X and return a transformed version.
+
+        Parameters
+        ----------
+        X : Series or Panel
+        y : Series or Panel
+        Returns
+        -------
+        transformed version of X
+        """
+        # input checks
+        if not isinstance(y, pd.DataFrame):
+            y = pd.DataFrame(y)
+
+        if not isinstance(X, pd.DataFrame):
+            X = pd.DataFrame(X)
```

Comment:
I am wondering about this type casting, should this not be covered by changing this given tags:
```
        "X_inner_mtype": [
            "pd.Series",
            "pd.DataFrame",
        ],  # which mtypes do _fit/_predict support for X?
        "y_inner_mtype": [
            "pd.Series",
            "pd.DataFrame",
        ], 
```
to just `pd.DataFrame`?
I guess @fkiraly should be able to answer this as well?

Reply:
seems to work, thanks for the hint